marketplace-service/src/main/java/com/axonivy/market/service/impl/ExternalDocumentServiceImpl.java

```diff
@@ -0,0 +1,103 @@
+package com.axonivy.market.service.impl;
+
+import com.axonivy.market.bo.Artifact;
+import com.axonivy.market.constants.DirectoryConstants;
+import com.axonivy.market.entity.ExternalDocumentMeta;
+import com.axonivy.market.entity.Product;
+import com.axonivy.market.factory.VersionFactory;
+import com.axonivy.market.repository.ExternalDocumentMetaRepository;
+import com.axonivy.market.repository.ProductRepository;
+import com.axonivy.market.service.ExternalDocumentService;
+import com.axonivy.market.service.FileDownloadService;
+import com.axonivy.market.util.MavenUtils;
+import com.axonivy.market.util.VersionUtils;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+
+@Log4j2
+@AllArgsConstructor
+@Service
+public class ExternalDocumentServiceImpl implements ExternalDocumentService {
+
+  private static final String DOC_URL_PATTERN = "/%s/index.html";
+  final ProductRepository productRepo;
+  final ExternalDocumentMetaRepository externalDocumentMetaRepo;
+  final FileDownloadService fileDownloadService;
+
+  @Override
+  public void syncDocumentForProduct(String productId, boolean isResetSync) {
+    productRepo.findById(productId).ifPresent(product -> {
+      var docArtifacts = Optional.ofNullable(product.getArtifacts()).orElse(List.of()).stream()
+          .filter(artifact -> BooleanUtils.isTrue(artifact.getDoc())).toList();
+      List<String> releasedVersions = Optional.ofNullable(product.getReleasedVersions()).orElse(List.of()).stream()
+          .filter(version -> VersionUtils.isValidFormatReleasedVersion(version)).toList();
```

Comment:
[Optional] Could you please convert to `version -> VersionUtils.isValidFormatReleasedVersion(version)` to `VersionUtils::isValidFormatReleasedVersion`